quark/script/frida/__init__.py

```diff
@@ -0,0 +1,197 @@
+            methodId = tuple(receivedEvent["identifier"])
+            self.watchedMethods.pop(methodId)
+
+
+@functools.lru_cache
+def _setupFrida(
+    appPackageName: str, protocol="usb", **kwargs: Any
+) -> Tuple[Device, FridaSession, int]:
+    device = None
+    if protocol == "usb":
+        device = frida.get_usb_device(**kwargs)
+    elif protocol == "local":
+        device = frida.get_local_device(**kwargs)
+    elif protocol == "remote":
+        device = frida.get_remote_device(**kwargs)
+
+    processId = device.spawn([appPackageName])
+    session = device.attach(processId)
+
+    return device, session, processId
+
+
+@functools.lru_cache
+def _injectAgent(frida: FridaSession) -> MethodCallEventDispatcher:
+    dispatcher = MethodCallEventDispatcher(frida)
+
+    pathToFridaAgentSource = pkg_resources.resource_filename(
+        "quark.script.frida", "agent.js"
+    )
+
+    with open(pathToFridaAgentSource, "r") as fridaAgentSource:
+        fridaAgent = dispatcher.frida.create_script(fridaAgentSource.read())
+        fridaAgent.on("message", dispatcher.receiveMessage)
+        fridaAgent.load()
+        dispatcher.script = fridaAgent
+
+    return dispatcher
+
+
+@dataclass
+class Behavior:
+    _message: Dict[str, str]
+
+    def hasString(self, pattern: str, regex: bool = False) -> List[str]:
+        """Check if the behavior contains strings
+
+        :param pattern: string to be checked
+        :param regex: True if the string is a regular expression, defaults to
+         False
+        :return: python list containing all matched strings
+        """
+        arguments = self.getParamValues()
+
+        allMatchedStrings = set()
+        for argument in arguments:
+            if regex:
+                matchedStrings = [
+                    match.group(0) for match in re.finditer(pattern, argument)
+                ]
+                allMatchedStrings.update(matchedStrings)
+            else:
+                if pattern in argument:
+                    return [pattern]
+
+        return list(allMatchedStrings)
+
+    def hasUrl(self) -> List[str]:
+        """Check if the behavior contains urls.
+
+        :return: python list containing all detected urls
+        """
+        return self.hasString(URL_REGEX, True)
+
+    def getParamValues(self) -> List[str]:
+        """Get parameter values from behavior.
+
+        :return: python list containing parameter values
+        """
+        return self._message["paramValues"]
+
+
+@dataclass
+class FridaResult:
+    _messageBuffer: List[str]
+
+    @property
+    def behaviorOccurList(self) -> List[Behavior]:
+        """List that stores instances of detected behavior in different part of
+         the target file.
+
+        :return: detected behavior instance
+        """
+        return [Behavior(message) for message in self._messageBuffer]
+
+
+def runFridaHook(
+    apkPackageName: str,
+    targetMethod: str,
+    methodParamTypes: str,
+    secondToWait: int,
```

Comment:
I think we can set a default number to `secondToWait`.
For example `secondToWait=10`.